std/traits.d

```diff
@@ -3340,6 +3341,171 @@ enum hasMember(T, string name) = __traits(hasMember, T, name);
     static assert(hasMember!(S, "foo"));
 }
 
+/++
+    Whether the symbol represented by the string, member, is a static member of
+    T.
+  +/
+template isStaticMember(T, string member)
+    if (__traits(hasMember, T, member))
+{
+    import std.meta : AliasSeq;
+    import std.traits : FunctionTypeOf;
+    alias sym = AliasSeq!(__traits(getMember, T, member))[0];
+
+    static if (is(typeof(sym) == function) || is(FunctionTypeOf!(typeof(&sym)) == function))
+        enum bool isStaticMember = __traits(isStaticFunction, sym);
+    else
+        enum bool isStaticMember = __traits(compiles, &sym);
```

Comment:
Feels like a weird recommendation. I wonder if there is a compiler workload difference?
I have a very poor feel for what things the compiler does quickly, and which are high-impact on compile time.